From 86ab2f1ff6b817f77f29452231409020390cd74f Mon Sep 17 00:00:00 2001
From: Kiran Kumar K <kirankumark@marvell.com>
Date: Mon, 6 Nov 2023 14:56:45 +0530
Subject: [PATCH 028/513] common/cnxk: fix issue with flow key alg for cnxk

Fix issue with flow key alg generation for cnxk device
in case of represented port.

Fixes: 840b8454a5ca ("drivers: add represented port flow item for cnxk")

Signed-off-by: Kiran Kumar K <kirankumark@marvell.com>
Reviewed-by: Jerin Jacob <jerinj@marvell.com>
Change-Id: I14b92092d654e6c296041e7e3ddcf27f4e04eb3d
---
 drivers/common/cnxk/roc_npc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/common/cnxk/roc_npc.c b/drivers/common/cnxk/roc_npc.c
index 72404c526cbe0..f8521b47aa8bc 100644
--- a/drivers/common/cnxk/roc_npc.c
+++ b/drivers/common/cnxk/roc_npc.c
@@ -1137,8 +1137,10 @@ npc_rss_action_program(struct roc_npc *roc_npc,
 	uint8_t alg_idx;
 	int rc;
 
-	if (flow->has_rep)
+	if (flow->has_rep) {
 		npc = roc_npc->rep_npc;
+		npc->flowkey_cfg_state = roc_npc->flowkey_cfg_state;
+	}
 
 	for (; actions->type != ROC_NPC_ACTION_TYPE_END; actions++) {
 		if (actions->type == ROC_NPC_ACTION_TYPE_RSS) {
-- 
2.25.1

